projects/fw_tutorial/inc/button.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+// Push button module for LED control written for the Tutorial board
+// This module is used as an introduction to Midnight Sun style firmware.
+// It is organized in such a way as to expose the user to basic fundamentals
+// and principals we follow when writing drivers and modules
+// Requires interrupts to be initialized
+#include "gpio.h"
+#include "gpio_it.h"
+
+// Type of Button
+typedef enum {
+  BUTTON_COLOUR_YELLOW = 0,
+  BUTTON_COLOUR_GREEN,
+  NUM_BUTTON_COLOURS,
+} ButtonColour;
+
+// Button settings structure
+typedef struct ButtonSettings {
+  GpioAddress button_addresses[NUM_BUTTON_COLOURS];  // Array of GPIO addresses for each push button
+  GpioAddress led_addresses[NUM_BUTTON_COLOURS];     // Array of GPIO addresses for each LED
```

Comment:
Any reason why not to use the debug LED definitions from `ms-helper`?

Reply:
tbh I didn't even know that existed 😅 I will take a look at that

Reply:
@karlding so the debug LED definitions are useful but I think they kinda defeat the purpose of portions of the module (e.g. the ability for new members to read the schematics and determine which GPIOs to use). So long-term, it might be better to remap these pins to other GPIOs so they are isolated from the microcontroller LEDs.